### PR TITLE
SPARKC-355: Shade Cassandra driver, Guava and Netty

### PR DESCRIPTION
**Pull request by pkolaczk**

### CODE REVIEW

**Comment by kovalenko-boris** [comment]
this pull request is very much needed, I run into issue described in SPARKC-355 Please fix it and merge in.

**Comment by webstergd** [comment]
I would also be a huge fan of this. This would fix the major headache I encounter when using tools like Zeppelin.

File: project/Settings.scala [hunk 1]
```diff
@@ -341,6 +341,14 @@ object Settings extends Build {
         case PathList("META-INF", xs @ _*) => MergeStrategy.last
         case x => old(x)
       }
+    },
+    assemblyShadeRules in assembly := {
+      val shadePackage = "shade.com.datastax.spark.connector"
+      Seq(
+        ShadeRule.rename("com.datastax.cassandra.driver.**" -> s"$shadePackage.driver.@1").inAll,
```
Comment (bchen-talend) [inline project/Settings.scala]: com.datastax.driver instead of com.datastax.cassandra.driver?

Comment (pkolaczk) [inline project/Settings.scala]: good catch

**Comment by RussellSpitzer** [comment]
Seen some issues with this

1) I see references to the Driver broken from some call to sc.cassandraTable using the assembly jar via --jars
2) User reported errors relating to commons-configuration (apache) references being shaded but the jar not be included in assembly. (Traced this down to the jar being brought in by Hadoop2ClientLIb but excluded since that is part of Spark which is excluded)

**Comment by pkolaczk** [comment]
I need to reopen this against b1.6

**Pull request closed without merging**